addons/File_Sender/src/presenter.js

```diff
@@ -0,0 +1,637 @@
+    presenter.views = [];
+    presenter.dialog = null;
+    presenter.pageIndex = -1;
+    presenter.$ICPage = null;
+    presenter.contextLoaded = false;
+
+    presenter.sentFileName = "";
+    presenter.sentFileUrl = "";
+
+    var currentScrollTop = 0;
+
+    var LessonSendFileType = {
+        'FILE': 0,
+        'AUDIO': 1,
+        'TEXT': 2,
+    };
+
+    var SOURCE_TYPES = {
+        'FILE': 'File',
+        'PARAGRAPH': 'Paragraph',
+        'MEDIA_RECORDER': 'Media Recorder'
+    };
+
+    var ERROR_CODES = {
+        V01: 'Source ID cannot be empty unless source type is set to File'
+    };
+
+    var sendFileEventType = "sendFile";
+
+
+    function presenterLogic(view, model, isPreview) {
+        presenter.addonID = model.ID;
+        presenter.$view = $(view);
+        presenter.isPreview = isPreview;
+
+        presenter.configuration = presenter.validateModel(model);
+        if (!presenter.configuration.isValid) {
+            presenter.$view.html(ERROR_CODES[presenter.configuration.errorCode]);
+            return;
+        }
+        presenter.loadViewHandlers();
+        presenter.attachHandlers();
+
+        presenter.views.sendFileButton.html(presenter.configuration.buttonText);
+        if (presenter.configuration.sourceType !== SOURCE_TYPES.FILE) {
+            presenter.views.form.css('display', 'none');
+            presenter.views.formLabel.css('display', 'none');
+            presenter.views.form.css('width', model['Width']+'px');
+            presenter.views.form.css('height', model['Height']+'px');
+        } else {
+            presenter.views.sendFileButton.css('width', model['Width']+'px');
+            presenter.views.sendFileButton.css('height', model['Width']+'px');
+        }
+
+        presenter.loadGlobalSendFileState();
+        presenter.hideSentFile();
+
+        if ((!isPreview) && (!presenter.contextLoaded)) {
+            presenter.hide();
+        }
+    }
+
+    presenter.show = function () {
+        presenter.$view.css("display", "");
+    }
+
+    presenter.hide = function () {
+        presenter.$view.css("display", "none");
+    }
+
+    presenter.validateModel = function (model) {
+        var sourceType = model.SourceType;
+        if (sourceType.length == 0) sourceType = SOURCE_TYPES.FILE;
+
+        if (sourceType !== SOURCE_TYPES.FILE && model.SourceId.length == 0) {
+             return {isValid: false, errorCode: 'V01'};
+        }
+
+        return {
+            ID: model.ID,
+            isValid: true,
+            sourceId: model.SourceId,
+            sourceType: sourceType,
+            buttonText: model.ButtonText,
+            dialogTitle: model.DialogTitle
+        }
+    }
+
+    presenter.setPlayerController = function(controller) {
+        presenter.playerController = controller;
+        presenter.pageIndex = presenter.playerController.getCurrentPageIndex();
+
+        presenter.loadContext(10);
+    };
+
+    presenter.loadContext = function(iterationsLeft) {
```

Comment:
if (iterationsLeft <= 0) {return;}
var context = blabla;
if (!context) {
    window.setTimeout(function(){presenter.loadContext(iterationsLeft-1)}, 500);
    return;
}
//... reszta kodu pod spodem - dzięki temu nie mamy zagnieżdżonych ifów i elsów, ale to raczej clean code niż samo działanie :D